src/perf_ringbuffer.c

```diff
@@ -245,12 +250,19 @@ perf_event_sample *hdr2samp(const struct perf_event_header *hdr,
     sample.ips = buf;
     buf += sample.nr;
   }
-  if (PERF_SAMPLE_RAW & mask) {}
+  if (PERF_SAMPLE_RAW & mask) {
+    // size_raw is a 32-bit integer!
+    sample.size_raw = ((flipper *)buf)->half[0];
+    sample.data_raw = sample.size_raw ? &((flipper *)buf)->half[1] : NULL;
+    buf += 1 + (sample.size_raw / sizeof(*buf)); // Advance + align
+  }
   if (PERF_SAMPLE_BRANCH_STACK & mask) {}
   if (PERF_SAMPLE_REGS_USER & mask) {
     sample.abi = *buf++;
-    // In case regs are not available, ignore this sample
-    if (sample.abi == PERF_SAMPLE_REGS_ABI_NONE) {
+    // Since some of the previous events are variable-length, ABI gives us
```

Comment:
Interesting !
Instrumentation telemetry would be nice here ! Perhaps we should leave markers of where we want to raise these events.